File: sdks/python/src/boxsdk/models/shared/collaborationallowlistexempttargets.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from boxsdk import utils
from ..shared import collaborationallowlistexempttarget as shared_collaborationallowlistexempttarget


@dataclass_json
@dataclasses.dataclass
class CollaborationAllowlistExemptTargets:
    r"""CollaborationAllowlistExemptTargets
    The part of an API response that describes marker
    based pagination
    """
    
    entries: Optional[list[shared_collaborationallowlistexempttarget.CollaborationAllowlistExemptTarget]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('entries') }})
    limit: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('limit') }})
    next_marker: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('next_marker') }})
    prev_marker: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('prev_marker') }})
    

File: ts-sdk/src/sdk/models/shared/collaborationallowlistexempttargets.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { CollaborationAllowlistExemptTarget } from "./collaborationallowlistexempttarget";



// CollaborationAllowlistExemptTargets
/** 
 * The part of an API response that describes marker
 * based pagination
**/
export class CollaborationAllowlistExemptTargets extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=entries", elemType: CollaborationAllowlistExemptTarget })
  entries?: CollaborationAllowlistExemptTarget[];

  @SpeakeasyMetadata({ data: "json, name=limit" })
  limit?: number;

  @SpeakeasyMetadata({ data: "json, name=next_marker" })
  nextMarker?: number;

  @SpeakeasyMetadata({ data: "json, name=prev_marker" })
  prevMarker?: number;
}
